examples_tests/60.ClusteredRendering/cull_common.glsl

```diff
@@ -0,0 +1,247 @@
+#ifndef CULL_COMMON_H
+#define CULL_COMMON_H
+
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_
+#error "_NBL_GLSL_WORKGROUP_SIZE_ must be defined"
+#endif
+
+#ifndef LIGHT_CONTRIBUTION_THRESHOLD
+#error "LIGHT_CONTRIBUTION_THRESHOLD must be defined"
+#endif
+
+#ifndef LIGHT_RADIUS
+#error "LIGHT_RADIUS must be defined"
+#endif
+
+#define INVOCATIONS_PER_LIGHT 8
+#define LIGHTS_PER_WORKGROUP (_NBL_GLSL_WORKGROUP_SIZE_/INVOCATIONS_PER_LIGHT)
+
+layout(local_size_x = _NBL_GLSL_WORKGROUP_SIZE_) in;
+
+#include <nbl/builtin/glsl/shapes/aabb.glsl>
+#include <nbl/builtin/glsl/math/quaternions.glsl>
+
+#include <../intersection_record.glsl>
+
+struct nbl_glsl_ext_ClusteredLighting_SpotLight
+{
+	vec3 position;
+	float outerCosineOverCosineRange;
+	uvec2 intensity;
+	uvec2 direction;
+};
+
+struct cone_t
+{
+	vec3 tip;
+	float height;
+	vec3 direction;
+	float cosHalfAngle;
+	float baseRadius;
+};
+
+cone_t getLightVolume(in nbl_glsl_ext_ClusteredLighting_SpotLight light)
+{
+	cone_t cone;
+
+	// tip
+	cone.tip = light.position;
+
+	// height
+	const float radiusSq = LIGHT_RADIUS * LIGHT_RADIUS;
+
+	const vec3 intensity = nbl_glsl_decodeRGB19E7(light.intensity);
+	const float maxIntensityComponent = max(max(intensity.r, intensity.g), intensity.b);
+	const float determinant = clamp(1.f - ((2.f * LIGHT_CONTRIBUTION_THRESHOLD) / (maxIntensityComponent * radiusSq)), -1.f, 1.f);
+
+	cone.height = LIGHT_RADIUS * inversesqrt(1.f / (determinant * determinant) - 1.f);
+
+	// direction
+	const vec2 dirXY = unpackSnorm2x16(light.direction[0]);
+	const vec2 dirZW = unpackSnorm2x16(light.direction[1]);
+	cone.direction = vec3(dirXY.xy, dirZW.x);
+
+	// cosHalfAngle
+	// Todo(achal): I cannot handle spotlights/cone intersection against AABB
+	// if it has outerHalfAngle > 90.f, hence the `max`
+	const float cosineRange = dirZW.y;
+	cone.cosHalfAngle = max(light.outerCosineOverCosineRange * cosineRange, 1e-3f);
+
+	// baseRadius
+	const float tanOuterHalfAngle = sqrt(max(1.f - (cone.cosHalfAngle * cone.cosHalfAngle), 0.f)) / cone.cosHalfAngle;
+	cone.baseRadius = cone.height * tanOuterHalfAngle;
+
+	return cone;
+}
+
+// Todo(achal): Rename to getClusterAABB
+nbl_glsl_shapes_AABB_t getCluster(in uvec3 localClusterID, in vec3 levelMinVertex, in float voxelSideLength)
+{
+	const vec3 camPos = pc.camPosGenesisVoxelExtent.xyz;
+
+	nbl_glsl_shapes_AABB_t cluster;
+	cluster.minVx = levelMinVertex + (localClusterID * voxelSideLength) + camPos;
+	cluster.maxVx = levelMinVertex + (localClusterID + uvec3(1u)) * voxelSideLength + camPos;
+	return cluster;
+}
+
+#define PLANE_COUNT 6
+vec4[PLANE_COUNT] getAABBPlanes(in nbl_glsl_shapes_AABB_t aabb)
```

Comment:
don't touch, but leave a comment saying its redundant and not needed

Reply:
actually function fully not needed.

Reply:
@nipunG314 remove this whole function before moving to `old_to_refactor` + put comment "function removed" on just its declaration